server/src/main/java/io/druid/segment/realtime/RealtimeManager.java

```diff
@@ -252,8 +252,14 @@ public void run()
           try {
             try {
               inputRow = firehose.nextRow();
+
+              if (inputRow == null) {
+                log.debug("thrown away null input row, considering unparseable");
```

Comment:
Like how many inputRows have been read up until this one? Yes, but would that actually be helpful? I would think it's not helpful and what you'd actually want is the kafka partition and offset. But there isn't really a way to get that at this point in the code, since it's buried inside the firehose.

That could potentially be solved by having Firehoses return InputRowAndMetadata (with two methods: getRow and getMetadata) where the metadata is something that we can toString in this error message. But, that's a Firehose interface change, which is tough to do since it's an external api.